mods/beds/api.lua

```diff
@@ -97,6 +100,19 @@ function beds.register_bed(name, def)
 			type = "fixed",
 			fixed = {0, 0, 0, 0, 0, 0},
 		},
+		on_destruct = function(pos)
+			local n = minetest.get_node_or_nil(pos)
+			if not n then return end
+			do_not_destruct = true
+			local dir = minetest.facedir_to_dir(n.param2)
+			-- Reverse direction
+			dir = {x = -dir.x, y = -dir.y, z = -dir.z}
+			local p = vector.add(pos,dir)
```

Comment:
Thanks for this method, I'm not very used to vector functions.
`do_no_destruct` and `vector.substract` will be fixed and pushed in about minutes.